vdom/src/com/vdom/core/PileCreator.java

```diff
@@ -0,0 +1,97 @@
+package com.vdom.core;
+
+import com.vdom.api.Card;
+
+import java.io.Serializable;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.LinkedHashMap;
+import java.util.Map;
+import java.util.Random;
+
+public abstract class PileCreator implements Serializable {
+    public abstract AbstractCardPile create(Card template, int count);
+}
+
+class DefaultCardPileCreator extends PileCreator {
+    public AbstractCardPile create(Card template, int count) {
+        return new SingleCardPile(template, count);
+    }
+}
+
+class RuinsPileCreator extends PileCreator {
+    public AbstractCardPile create(Card template, int count) {
+
+        Map<Card, Integer> cards = new HashMap<Card, Integer>();
```

Comment:
This selection logic is rather obtuse. I can't easily tell if this follows the rules for creating the Ruins pile - which is that the 50 Ruins are all shuffled and then the appropriate amount (count) are taken from that shuffled pile. This leads to some weird circumstances where you have games without a particular kind of Ruins, but you can never have more than 10 of a particular kind.